### DISCUSSION TITLE
Snap7 on Mac (M1)

### DISCUSSION THREAD

**Discussion by aj-88** [gijzelaerr/python-snap7#338] [Q&A] [answered]
Hello, 

I'm trying to install Snap7 on my Mac (M1pro).
I installed python-snap7 via the Binary wheel. This install was completed, but it can't find the snap7 library. On my other (windows) laptop I had to add a dll file to a specific folder. After that it worked, but on my Mac I can't find a solution for the missing library problem. I followed the snap7 install for Mac, but for architecture you can only choose i386 or x86_64. I chose the x86_64. But it didn't solve the problem.

My very simple program to test:
`import snap7
plc = snap7.client.Client()`

Gives the following error:

`Traceback (most recent call last):
  File "/Users/user1/Desktop/test.py", line 3, in <module>
    plc = snap7.client.Client()
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.9/site-packages/snap7/client.py", line 66, in __init__
    self._library = load_library(lib_location)
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.9/site-packages/snap7/common.py", line 75, in load_library
    return Snap7Library(lib_location).cdll
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.9/site-packages/snap7/common.py", line 66, in __init__
    raise Snap7Exception(msg)
snap7.exceptions.Snap7Exception: can't find snap7 library. If installed, try running ldconfig`

I hope you guys can help me out.
Thanks in advance!

**Comment by swamper123** [comment]
Hey ho,

we are on to it and it will be released in the near future.
So stay tuned for updates. :)

**Comment by nikteliy** [comment]
@aj-88  If you don't want to wait, you can install the package manually from the latest build:
1. download [the latest build](https://github.com/gijzelaerr/python-snap7/suites/4492230411/artifacts/119739119)
2. unzip wheels.zip
3. run `python3 -m pip install /path/to/wheels/macOS/python_snap7-1.1-py3-none-macosx_10_9_universal2.whl`

**Reply by aj-88** [reply]
Thanks, I will try this today or tomorrow. :)

**Reply by aj-88** [reply]
It's working now, thanks a lot!

**Comment by gijzelaerr** [comment, marked as answer]
i've just  uploaded `python_snap7-1.1-py3-none-macosx_10_9_universal2.whl` to pypi.
